### PR TITLE
[IMP] account: add a default currency on bank/cash journals

### PR DESCRIPTION
**Pull request by aliyatastemirova**
Setting company's currency on bank/cash journals, if they are not set
by the user, helps avoid issues described in Task 2834678.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/account/models/account_journal.py [hunk 1]
```diff
@@ -104,7 +104,8 @@ def _default_invoice_reference_model(self):
     invoice_reference_model = fields.Selection(string='Communication Standard', required=True, selection=[('odoo', 'Odoo'), ('euro', 'European')], default=_default_invoice_reference_model, help="You can choose different models for each type of reference. The default one is the Odoo reference.")
 
     #groups_id = fields.Many2many('res.groups', 'account_journal_group_rel', 'journal_id', 'group_id', string='Groups')
-    currency_id = fields.Many2one('res.currency', help='The currency used to enter statement', string="Currency")
+    currency_id = fields.Many2one('res.currency', help="- Currency used to enter Bank or Cash statements.\n"
+                                  "- Default currency of each new move on all other journal types", string="Currency", compute='_compute_journal_currency', store=True, readonly=False, required=True, precompute=True)
```
Comment (smetl) [inline addons/account/models/account_journal.py:108]: this indentation is chaotic
```py
currency_id = fields.Many2one(
    comodel_name='res.currency',
    string="Currency",
    compute='_compute_journal_currency',
    store=True,
    readonly=False,
    required=True,
    precompute=True,
    help="The default currency suggested on journal entries or the currency for bank transactions",
)
```
(changed the helper, what you do think?)

File: addons/account/models/account_journal.py [hunk 2]
```diff
@@ -382,6 +383,12 @@ def _compute_alias_name(self):
         for journal in self:
             journal.alias_name = journal.alias_id.alias_name
 
+    @api.depends('company_id')
+    def _compute_journal_currency(self):
+        for journal in self:
+            if not journal.currency_id:
+                journal.currency_id = journal.company_id.currency_id
```
Comment (smetl) [inline addons/account/models/account_journal.py:390]:
```suggestion
    @api.depends('company_id')
    def _compute_journal_currency(self):
        for journal in self:
            journal.currency_id = journal.company_id.currency_id
```

File: addons/account/models/account_move.py [hunk 3]
```diff
@@ -503,7 +502,7 @@ def _onchange_invoice_date(self):
 
     @api.onchange('journal_id')
     def _onchange_journal(self):
-        if self.journal_id and self.journal_id.currency_id:
+        if self.journal_id:
```
Comment (smetl) [inline addons/account/models/account_move.py:505]:
```suggestion
    new_currency = self.journal_id.currency_id
    if new_currency and new_currency != self.currency_id:
        ...
```

File: addons/account/models/chart_template.py [hunk 4]
```diff
@@ -386,7 +386,7 @@ def _create_bank_journals(self, company, acc_template_ref):
                 'name': acc['acc_name'],
                 'type': acc['account_type'],
                 'company_id': company.id,
-                'currency_id': acc.get('currency_id', self.env['res.currency']).id,
+                'currency_id': acc.get('currency_id', self.env['res.currency']).id or company.currency_id.id,
                 'sequence': 10,
             })
```
Comment (smetl) [inline addons/account/models/chart_template.py:391]:
```py
            journal_vals = {
                'name': acc['acc_name'],
                'type': acc['account_type'],
                'company_id': company.id,
                'sequence': 10,
            }
            if acc.get('currency_id'):
                journal_vals['currency_id'] = acc['currency_id']
```

File: addons/account/models/account_journal.py (hunk 1 above)
Comment (aliyatastemirova) [reply addons/account/models/account_journal.py:108]: I think this helper is ok, I copied what @tsb-odoo wrote to me :slightly_smiling_face:

**Comment by aliyatastemirova** [comment]
The task will be canceled or the spec will be changed.

**Pull request closed without merging**